documentation/tut/toy-example-reporter.rst

```diff
@@ -0,0 +1,241 @@
+
+
+Compute headroom
+-------------------
+
+In this case, the discharge headroom is nothing but the difference between the grid connection capacity and the PV power. To compute that quantity, we can use the `AggregatorReporter` using the weights to make the PV to substract the grid connection capacity.
+
+In practice, we need to create the `config` and `parameters`:
+
+.. code-block:: bash
+
+    $ echo "
+    $ {
+    $    'weights' : {
+    $        'grid connection capacity' : 1.0,
+    $        'PV' : -1.0,
+    $    }
+    $ }" > headroom-config.json
+
+
+.. code-block:: bash
+
+    $ echo "
+    $ {
+    $     'input' : [{'name' : 'grid connection capacity','sensor' : 7},
+    $                {'name' : 'PV', 'sensor' : 3}],
+    $     'output' : [{'sensor' : 8}]
+    $ }" > headroom-parameters.json
+
+
+Finally, we can create the reporter with the following command:
+
+.. code-block:: bash
+
+    $ TOMORROW=$(date --date="next day" '+%Y-%m-%d')
+    $ flexmeasures add report --reporter AggregatorReporter \
+       --parameters headroom-parameters.json --config headroom-config.json \
+       --start-offset DB,1D --end-offset DB,2D \
+       --resolution PT15M
+
+Now we can visualize the headroom in the following `link <http://localhost:5000/sensor/8/>`_, it should resemble the following image.
+
+.. image:: https://github.com/FlexMeasures/screenshots/raw/main/tut/toy-schedule/sensor-data-headroom.png
+    :align: center
+|
+
+The graph shows that the capacity of the grid is at full disposal for the battery when there's no sun (thus no PV generation), while
+at noon the battery can only discharge at 280kW max.
+
+Process scheduler profit
+-------------------------
+
+For the second part of this tutorial, we are going to use the `ProfitLossReporter` to compute the losses (defined as `cost - revenue`) of operating the 
+process from Tut. Part III, under the three different policies: INFLEXIBLE, BREAKABLE and SHIFTABLE.
+
+In addition, we'll explore another way to invoke reporters: data generators. Without going too much into detail, data generators
+create new data. The thee main types are: `Reporters`, `Schedulers` and `Forecasters`. This will come handy as the three reports that
+we are going to create share the same `config`. The `config` defines the price sensor to use and sets the reporter to work in **losses** mode which means
+that it will return positive values cost and negative values revenue.
+
+Still, we need to define the parameters. The three reports share the same structure for the paramters with the following fields:
+* `input`: sensor that stores the power/energy flow. The number of sensors is limit to 1.
+* `output`: sensor to store the result of the report. We can provide sensors with differen resolutions to store the same results at different time scales.
+
+It's possible to define the `config` and `parameters` in JSON or YAML formats.
```

Comment:
Another candidate for a `.. note::`. It's a side-note in the context of the tutorial.